Sources/FoundationInternationalization/Formatting/Date/ICUDateFormatter.swift

```diff
@@ -191,6 +191,24 @@ final class ICUDateFormatter {
         })
     }
 
+    // MARK: - Getting symbols
+
+    func symbols(for key: UDateFormatSymbolType) -> [String] {
+        let symbolCount = udat_countSymbols(udateFormat, key)
```

Comment:
It looks like we cache the formatter, but do we also need to cache this result? Calling it in a loop and checking if it's cached inside ICU itself would be a good idea.

Reply:
I thought about it too but didn't pursue because it was already much better than what we had before, at which time we didn't even cache the formatter. But now that you mentioned it we might as well do it while we're here

Reply:
I'm actually not seeing any stack trace inside ICU. It looks like the counts and the symbols are loaded as constants upon initializing a `udat`, i.e. the known-to-be-expensive operation. The most expensive part of this function lies in `_withResizingUCharBuffer` due to utf-16 to utf-8 transcoding. I think we can forgo this cache here.